data/data/bootstrap/baremetal/files/usr/local/bin/master-bmh-update.sh

```diff
@@ -30,3 +30,27 @@ while ! oc annotate --overwrite -n openshift-machine-api baremetalhosts --all "b
     sleep 5
     echo "Unpause failed, retrying"
 done
+
+# Pause control-plane management via ironic for Two-Node OpenShift with Fencing (TNF), since this conflicts with how TNF achieves HA
+TNF_TOPOLOGY="DualReplica"
+
+echo "Looking up control-plane topology"
+CONTROL_PLANE_TOPOLOGY=$(oc get infrastructures.config.openshift.io cluster -o json | jq -r '.status.controlPlaneTopology')
+if [ -z "${CONTROL_PLANE_TOPOLOGY}" ]; then
+    echo "Error: couldn't lookup control-plane topology" >&2
+    exit 1
+elif [ "${CONTROL_PLANE_TOPOLOGY}" = "${TNF_TOPOLOGY}" ]; then
+    echo "Control-plane topology set to '${TNF_TOPOLOGY}'; setting the control-plane hosts to detached"
+    while ! oc annotate --overwrite bmh -n openshift-machine-api -l installer.openshift.io/role=control-plane baremetalhost.metal3.io/detached="" ; do
+        sleep 5
+        echo "Setting control-plane nodes to detached failed, retrying"
+    done
+
+    while [ "$(oc get bmh -n openshift-machine-api -l installer.openshift.io/role=control-plane -o json | jq '.items[].status.operationalStatus' | grep detached -c)" -lt "$CONTROL_PLANE_REPLICA_COUNT"  ]; do
+        echo "Waiting for operationalStatus of all masters to become detached"
+        oc get bmh -n openshift-machine-api -o=custom-columns=NAME:.metadata.name,"OPERATIONAL STATUS":.status.operationalStatus || true
+        sleep 20
```

Comment:
It's fine to wait for this I guess, but we're not actually blocking anything at this point. Once the ironic service is stopped on line 23, the VIP can fall over to the cluster control plane and the cluster's metal³ can start. If we are trying to prevent BMO on the host from _ever_ reconciling the control plane BMHs then there is a race here.

Reply:
Good point. I think it should be fine to detach the hosts before stopping Ironic, although in this case the process will be slightly longer

Reply:
In fact, waiting for the detached state will not work here (sorry for confusion) because BMO is down already.

Reply:
I workshopped this section with @dtantsur in https://redhat-internal.slack.com/archives/CFP6ST0A3/p1744720045125349?thread_ts=1744635159.894649&cid=CFP6ST0A3.

We decided to move the detach block before shutting down ironic as well as removing the wait entirely. Please let me know if that introduces any new concerns.